### DISCUSSION TITLE
feat: add ydb client

### DISCUSSION THREAD

**Discussion by maxbronnikov10** [mikro-orm/mikro-orm#5770] [Ideas]
Two years ago a new database was released - ydb (https://github.com/ydb-platform).  It is SQL like, but it provides sharding functionality out of the box.

I would like to use this database in microorm and am willing to write the corresponding funcionals.

After looking through the repository code, I saw that most of the drivers are based on the knex.js library.

The knex.js library itself says that it doesn't make sense to add a new dialect. But anyway I have two solutions to the problem:
1. implement all the functionality in this repository
2. try to implement the dialect in knex and add it here in the future

For me more preferable variant is to implement everything here, as you are more actively considering issues compared to knex.js.

I would be grateful for feedback!

**Comment by B4nan** [comment]
I would prefer if this is first born as a 3rd party driver, I don't mind adding more extension points (or improving those we have now) if needed.

If its SQL based, I would strongly recommend you try to create a knex dialect for it (so you can reuse what the `@mikro-orm/knex` package is doing, which is _a lot_), otherwise it will be quite complex to implement all the necessary functionality yourself (but still possible even from outside).

> For me more preferable variant is to implement everything here, as you are more actively considering issues compared to knex.js.

You should create custom knex dialect, not PR it to knex directly, I agree that wouldn't probably be successful. Last time when a user faced some issue with knex and I told him to try to PR it, they ignored him completely and I ended up adding his patches to our custom dialect :] 

Which is indeed possible, we have things like libsql and mariadb, those are not officially supported knex dialects too.

**Reply by maxbronnikov10** [reply]
I want to wrap the ydb-sdk(https://www.npmjs.com/package/ydb-sdk) package and implement all necessary functionality accordingly.

Ok, I will implement knex dialect here, do I need to separate somehow PR, conventionally first PR for dialect implementation and second one for mikroorm features implementation. Or can I do everything in one PR?

**Reply by B4nan** [reply]
As I said, I would like to see this as 3rd party package, so no PRs here, you create a new repository and publish it to NPM yourself.

**Reply by B4nan** [reply]
For a dialect to be added to the ORM repository I would like to see complete support including schema diffing and a complete test coverage, that is a huge task to accomplish, publishing it first on your own can happen even with an early version that handles only runtime.

**Reply by maxbronnikov10** [reply]
Okay, thx for feedback
